.github/workflows/cleanup.yml

```diff
@@ -0,0 +1,140 @@
+              -H "Accept: application/vnd.github.v3+json" \
+              -X DELETE \
+              repos/${{ github.repository }}/actions/runs/$ID \
+            | echo;
+          done
+          echo "Done"
+
+  ## Use the gh cli and API to delete actions that have been failed/skipped for at least x days
+  ##   - currently set for: 1 days (EPOCH_1)
+  delete-failed-actions:
+    name: Delete Failed Workflows
+    runs-on: ubuntu-latest
+    steps:
+      - name: Github API Call
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: |
+          TS=$(date --iso-8601=ns -u -d @$((EPOCHSECONDS-${{ env.EPOCH_1 }})))
+          echo "Deleting all failed/skipped actions older than: ${TS}"
+          gh extension install actions/gh-actions-cache
+          gh api \
+            --paginate \
+            -H "Accept: application/vnd.github+json" \
+            repos/${{ github.repository }}/actions/runs |
+          for ID in `
+            jq -r --arg TS "$TS" '
+              .workflow_runs[]
+              | { id, updated_at,status,conclusion } 
+              | select((.updated_at < $TS) and (.conclusion != "success" and .status == "completed"))
+              | {id}
+              | .[]
+            '
+          `; do
+            echo "*** Deleting action: $ID (https://github.com/${{ github.repository }}/actions/runs/${ID})"
+            gh api \
+              -H "Accept: application/vnd.github.v3+json" \
+              -X DELETE \
+              repos/${{ github.repository }}/actions/runs/$ID \
+            | echo;
+          done
+          echo "Done"
+
+  ## Use the gh cli and API to delete caches that have existed for at least 1 days
+  ##   - currently set for: 1 days (EPOCH_1)
+  ##   - ignore caches with a key of `stacks-blockchain-bitcoin-binaries`
+  delete-old-caches:
+    # if: ${{ false }}
+    name: Delete Caches
+    runs-on: ubuntu-latest
+    steps:
+      - name: Github API Call
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: |
+          TS=$(date --iso-8601=ns -u -d @$((EPOCHSECONDS-${{ env.EPOCH_1 }})))
+          echo "Deleting all caches older than: ${TS}"
+          gh extension install actions/gh-actions-cache
+          gh api \
+            --paginate \
+            -H "Accept: application/vnd.github+json" \
+            repos/${{ github.repository }}/actions/caches |
+          for ID in `
+            jq -r --arg TS "$TS" '
+              .actions_caches[]
+              | { id, last_accessed_at, key }
+              | select((.last_accessed_at < $TS) and (.key != "stacks-blockchain-bitcoin-binaries" ))
+              | {id}
+              | .[]
+            '
+          `; do
+            echo "*** Deleting cache: $ID"
+            gh api \
+              -H "Accept: application/vnd.github.v3+json" \
+              -X DELETE \
+              repos/${{ github.repository }}/actions/caches/$ID \
+            | echo;
+          done
+          echo "Done"
```

Comment:
Personally, I'd love for there to be comments within this section - I'm having a hard time uderstanding the individual commands even with the "echos"

Reply:
will do!

Reply:
comments to the shell being run here have been added